lib/ui.rb

```diff
@@ -2,13 +2,17 @@
 
 class UI
```

Comment:
There's a lot of duplication in this class. Almost every method writes and reads. Can you think of a way to minimise this duplication?